arangod/Pregel/Statistics.h

```diff
@@ -87,11 +88,8 @@ struct StatsManager {
     }
   }
 
-  void accumulateMessageStats(VPackSlice data) {
-    VPackSlice sender = data.get(Utils::senderKey);
-    if (sender.isString()) {
-      _serverStats[sender.copyString()].accumulate(data);
-    }
+  void accumulateMessageStats(std::string const& senderId, VPackSlice data) {
+    _serverStats[senderId].accumulate(data);
```

Comment:
I think we should be using `.at()` (which will throw an exception and not just crash if we don't know the server ID), or even have error handling here (which might be an option for later us).
```suggestion
    _serverStats.at(senderId).accumulate(data);
```

Reply:
I have to investigate here: Now, an exception is thrown...